test/object-store/util/sync/redirect_server.hpp

```diff
@@ -24,28 +24,52 @@
 #include <realm/sync/network/network.hpp>
 #include <realm/sync/network/websocket.hpp>
 #include <realm/util/future.hpp>
+#include <realm/util/random.hpp>
 #include <realm/util/uri.hpp>
 #include <external/json/json.hpp>
 
-#include <catch2/catch_all.hpp>
-
+#include <iostream>
 #include <thread>
 
 namespace realm::sync {
 
 class RedirectingHttpServer {
 public:
+    enum Event {
+        error,
+        location,
+        redirect,
+        ws_redirect,
+    };
+
+    // Allow the redirecting server to choose the listen port
     RedirectingHttpServer(std::string redirect_to_base_url, std::shared_ptr<util::Logger> logger)
-        : m_redirect_to_base_url(std::move(redirect_to_base_url))
+        : RedirectingHttpServer(std::move(redirect_to_base_url), 0, std::move(logger))
+    {
+    }
+
+    // Specify the listen port to use
+    RedirectingHttpServer(std::string redirect_to_base_url, uint16_t listen_port,
+                          std::shared_ptr<util::Logger> logger)
+        : m_redirect_to_base_url(trim_url(redirect_to_base_url))
+        , m_redirect_to_base_wsurl(make_wsurl(m_redirect_to_base_url))
         , m_logger(std::make_shared<util::PrefixLogger>("HTTP Redirector ", std::move(logger)))
         , m_acceptor(m_service)
         , m_server_thread([this] {
             m_service.run_until_stopped();
         })
     {
-        m_acceptor.open(m_endpoint.protocol());
-        m_acceptor.bind(m_endpoint);
-        m_endpoint = m_acceptor.local_endpoint();
+        reset_state();
+        network::Endpoint ep;
+        if (listen_port != 0) {
+            // If a port is provided, then configure the endpoint with this port num
+            ep = network::Endpoint(network::make_address("0.0.0.0"), listen_port);
```

Comment:
why do we need to supply an explicit listen port?

Reply:
It was for the redirect server command line tool - removed.